src/include/catalog/database_handle.h

```diff
@@ -0,0 +1,146 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "catalog/attr_def_handle.h"
+#include "catalog/attribute_handle.h"
+#include "catalog/catalog.h"
+#include "catalog/class_handle.h"
+#include "catalog/namespace_handle.h"
+#include "catalog/type_handle.h"
+#include "storage/sql_table.h"
+#include "transaction/transaction_context.h"
+
+namespace terrier::catalog {
+
+class Catalog;
+class AttributeHandle;
+class AttrDefHandle;
+class NamespaceHandle;
+class TypeHandle;
+struct SchemaCol;
+
+/**
+ * A database entry represents a row in pg_database catalog.
+ */
+class DatabaseEntry : public CatalogEntry<db_oid_t> {
+ public:
+  /**
+   * Constructor
+   * @param oid database def oid
+   * @param sql_table associated with this entry
+   * @param entry a row in pg_database that represents this table
+   */
+  DatabaseEntry(db_oid_t oid, catalog::SqlTableRW *sql_table, std::vector<type::TransientValue> &&entry)
+      : CatalogEntry(oid, sql_table, std::move(entry)) {}
+};
+
+/**
+ * A DatabaseHandle provides access to the (global) system pg_database
+ * catalog.
+ *
+ * This pg_database is a subset of Postgres (v11)  pg_database, and
```

Comment:
We should bring in all of `pg_database` if possible.

Reply:
1. Deleted obsolete portion of comment
2. Added in remaining fields of pg_database.